docs/src/pages/components/steppers/CustomizedSteppers.js

```diff
@@ -36,12 +36,12 @@ const QontoConnector = styled(StepConnector)(({ theme }) => ({
   },
 }));
 
-const QontoStepIconRoot = styled('div')(({ theme, styleProps = {} }) => ({
+const QontoStepIconRoot = styled('div')(({ theme, styleProps }) => ({
```

Comment:
But where is `styleProps` coming from? It's not injected by `styled`, right? Instead it is passed from the callsite so the component should declare it, not `styled`.

Reply:
Yes, the component is declaring it https://github.com/mui-org/material-ui/blob/c9ba9cff395e1dc74821e57ae6f3d3186319a575/docs/src/pages/components/steppers/CustomizedSteppers.js#L64

I don't understand the question.

Reply:
> Yes, the component is declaring it

But we can't express this at a type level as far as I know. So `styled('div')(({ styleProps }) => ({}))` should really just throw because we don't know the shape of `styleProps`.

Wouldn't  `const Div = styled('div')(({ styleProps }) => ({}))` also allow `<Div />`?

Reply:
I probably again don't understand the comment, but I am not sure why we are discussing this for the JavaScript implementation. Should we add propTypes to make this required prop? This is transpiled from the TS implementation, this is why the line is changed. In the TS implementation, it is declared as:

```tsx
const QontoStepIconRoot = styled('div')<{ styleProps: { active?: boolean } }>(({ theme, styleProps }) => ({ ... });
```

And with this the `styleProps` is required field. This is an inner component, and we can make sure the `styleProps` are always passed.

Reply:
I did a mistake with the response. We treat the `styleProps` as a required field, as the parent component of the `QontoStepIconRoot` will always pass them. Does it makes sense now?

Reply:
> it is passed from the callsite so the component should declare it, not styled.

I don't get this part. The `styled` is asking for a specific set of props to apply the styles, TypeScript should check that the callsite is providing the required `stylesProps` prop. `styled` swallow the prop so AFAIK, the component shouldn't change.

Reply:
> TypeScript should check that the callsite is providing the required stylesProps prop to styled

Where is the shape for the `styleProps` declared?

Reply:
https://github.com/mui-org/material-ui/pull/26461/files#diff-96ac08f7944111fabb40f37b9d302baf2760434e084914e0fc3e46b02a768c3cR39 here

Reply:
> Where is the shape for the styleProps declared?

In this very instance? We comment on a JavaScript file: nowhere.

In general, I assume where we create a new component with `styled(WrappedComponent)(myStyles)`?

Reply:
This is a JS file :facepalm: